solidity/contracts/bridge/Bridge.sol

```diff
@@ -1918,28 +1923,81 @@ contract Bridge is Ownable, EcdsaWalletOwner {
-    //          given wallet by request's redeemable amount computed as
-    //          `requestedAmount - treasuryFee`.
-    //       8. Call `wallets.notifyRedemptionTimedOut` to propagate timeout
-    //          consequences to the wallet.
+    /// @notice Notifies that there is a pending redemption request associated
+    ///         with the given wallet, that has timed out. The redemption
+    ///         request is identified by the key built as
+    ///         `keccak256(walletPubKeyHash | redeemerOutputScript)`.
+    ///         The results of calling this function: the pending redemptions
+    ///         value for the wallet will be decreased by the requested amount
+    ///         (minus treasury fee), the tokens taken from the redeemer on
+    ///         redemption request will be returned to the redeemer, the request
+    ///         will be moved from pending redemptions to timed-out redemptions.
+    ///         If the state of the wallet is `Live` or `MovingFunds`, the
+    ///         wallet operators will be slashed.
+    ///         Additionally, if the state of wallet is `Live`, the wallet will
+    ///         be closed or marked as `MovingFunds` (depending on the presence
+    ///         or absence of the wallet's main UTXO) and the wallet will no
+    ///         longer be marked as the active wallet (if it was marked as such).
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    /// @param redeemerOutputScript  The redeemer's length-prefixed output
+    ///        script (P2PKH, P2WPKH, P2SH or P2WSH)
+    /// @dev Requirements:
+    ///      - The redemption request identified by `walletPubKeyHash` and
+    ///        `redeemerOutputScript` must exist
+    ///      - The amount of time defined by `redemptionTimeout` must have
+    ///        passed since the redemption was requested (the request must be
+    ///        timed-out).
+    function notifyRedemptionTimeout(
+        bytes20 walletPubKeyHash,
+        bytes calldata redeemerOutputScript
+    ) external {
+        uint256 redemptionKey = uint256(
+            keccak256(abi.encodePacked(walletPubKeyHash, redeemerOutputScript))
+        );
+        RedemptionRequest storage request = pendingRedemptions[redemptionKey];
+
+        require(request.requestedAt > 0, "Redemption request does not exist");
+        require(
+            /* solhint-disable-next-line not-rely-on-time */
+            request.requestedAt + redemptionTimeout < block.timestamp,
+            "Redemption request has not timed out"
+        );
+
+        // Update the wallet's pending redemptions value
+        Wallets.Wallet storage wallet = wallets.registeredWallets[
+            walletPubKeyHash
+        ];
+        wallet.pendingRedemptionsValue -=
+            request.requestedAmount -
+            request.treasuryFee;
+
+        // Return the requested amount of tokens to the redeemer
+        bank.transferBalance(request.redeemer, request.requestedAmount);
```

Comment:
We read the `request` as a pointer (and not the copy) to `pendingRedemption` element:
```
RedemptionRequest storage request = pendingRedemptions[redemptionKey];
```

And then we delete, so the `request` is zeroed:
```
delete pendingRedemptions[redemptionKey];
```

So we cannot move 
```
bank.transferBalance(request.redeemer, request.requestedAmount);
```
after deletion.

I would either leave it as is or work on a copy of the `request`.

Reply:
Moved the `bank.transferBalance(...)` at the end of the function (but before the even emitting).
I had to copy the request (use `memory` instead of `storage`).
 It decreased the gas consumption of `notifyRedemptionTimeout` a little bit (~1k).